### PR TITLE
Recreating search

### PR DESCRIPTION
**Pull request by JuliusKarl**
Significant Changes

- Search Bar Service
- Search Page

### CODE REVIEW

File: research-hub-web/src/app/components/search-bar/search-bar.component.html [hunk 1]
```diff
@@ -2,11 +2,15 @@
...
       <mat-form-field fxFlex floatLabel="never">
         <input role="textbox" #searchTextInput matInput placeholder="Search" [(ngModel)]="searchText"
-          (ngModelChange)="searchBarService.setCurrentPage(1); searchBarService.createResultsList()">
+          (keydown.enter)="searchBarService.setCurrentPage(1); searchBarService.createResultsList()">
         <button matSuffix role="button" aria-label="close" *ngIf="searchText" (click)="clearSearchText();this.searchBarService.createResultsList()"
           (keydown.enter)="clearSearchText();this.searchBarService.createResultsList()" mat-icon-button>
           <mat-icon>close</mat-icon>
         </button>
       </mat-form-field>
+      <button matSuffix role="button" aria-label="search" (click)="clearSearchText();this.searchBarService.createResultsList()"
```
Comment (rosemcc) [inline research-hub-web/src/app/components/search-bar/search-bar.component.html:11]: remove clearSearchText() ?

File: research-hub-web/src/app/components/search-page/search-page.component.html [hunk 2]
```diff
@@ -112,7 +112,7 @@ <h3 style="margin-bottom: 0" class="search-results-text">{{ this.searchBarServic
 
         <!-- No results message -->
         <div *ngIf="searchBarService.getTotalPages() == 0" class="no-results">
-            <p>Sorry, your search for "<i><b>{{ searchBarService.getSearchText() }}</b></i>" did not match anything on the ResearchHub.</p>
+            <p>Sorry, your search for "<i><b>{{ searchBarService.getSearchText() }}</b></i>" in All Categories, did not match anything on the ResearchHub.</p>
```
Comment (rosemcc) [inline research-hub-web/src/app/components/search-page/search-page.component.html:115]: remove 'in all categories'

**Comment by rosemcc** [comment]
e2e failing on a few tests

**Comment by rosemcc** [comment]
Search is working a lot nicer with it just searching after pressing enter. 
I still feel like it needs a loading indicator while it's searching, especially for slow connections

**Comment by JuliusKarl** [comment]
> Search is working a lot nicer with it just searching after pressing enter.
> I still feel like it needs a loading indicator while it's searching, especially for slow connections

Search loading indication has been added

File: research-hub-web/src/app/components/search-bar/search-bar.service.ts [hunk 3]
```diff
@@ -222,11 +222,15 @@ export class SearchBarService {
...
       let categories = this.getCategory().map(x => { return  x });
+      console.log("Page types before event: ", categories);
+      console.log("Categories before event: ", pageTypes);
 
       // If event is selected, remove it from search parameters (will be manually handled below)
       if (this.getCategory().includes(this.getEventId())) {
         categories.splice(this.getCategory().indexOf(this.getEventId()), 1)
         pageTypes = ["event"]
+        console.log("Page types event: ", pageTypes);
+        console.log("Categories after event: ", categories);
```
Comment (rosemcc) [inline research-hub-web/src/app/components/search-bar/search-bar.service.ts:233]: maybe remove console logs if you don't need them

File: research-hub-web/src/app/components/search-page/search-page.component.html [hunk 4]
```diff
@@ -109,17 +119,17 @@ <h3 style="margin-bottom: 0" class="search-results-text">{{ this.searchBarServic
...
-                Let us know what you were looking for via our <a href="{{ feedbackUrl }}">Feedback page</a><br>
-                You can try the Staff Service Centre in the <a href="{{ staffIntranet }}">Staff Intranet</a><br>
-            </p>
-        </div>
         <app-collection-list *ngIf="searchBarService.getTotalPages() != 0" [type]="'search'" [collection]="allCurrentPages"></app-collection-list>
     </div>
+
+    <!-- No results message -->
+    <div *ngIf="searchBarService.getTotalPages() == 0" class="no-results">
+        <p>Sorry, your search for "<i><b>{{ searchBarService.getSearchText() }}</b></i>" in All Categories, did not match anything on the ResearchHub.</p>
```
Comment (rosemcc) [inline research-hub-web/src/app/components/search-page/search-page.component.html:127]: remove "in All Categories"

**Comment by rosemcc** [comment]
search-page.component.html: In search Order By options can you please change 'Default' to 'Relevance'

**Comment by rosemcc** [comment]
Hey Julius, just two comments above to be resolved, then this is good to go

**Comment by rosemcc** [comment]
Oh wait, one more thing I think I already mentioned it in another comment but the 'search' icon doesn't behave as I would expect. It clears the search text and gives you all results instead of what I would expect which would be to keep the search text there and execute the search.. if this comment doesn't make sense, let me know :)

**Comment by rosemcc** [comment]
> Oh wait, one more thing I think I already mentioned it in another comment but the 'search' icon doesn't behave as I would expect. It clears the search text and gives you all results instead of what I would expect which would be to keep the search text there and execute the search.. if this comment doesn't make sense, let me know :)

In other words, make it so that clicking the search icon (the magnifying glass) doesn't clear the search text

**Comment by rosemcc** [comment]
Maybe you fixed this stuff already in Hotfix - if so, ignore my comments :)

**Pull request merged**